conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java

```diff
@@ -162,10 +164,7 @@ public static <T> Set<T> newNonNullSet() {
 
     public static <T> Set<T> newNonNullSet(Iterable<? extends T> iterable) {
         Set<T> set = newSet(iterable);
-        for (T item : set) {
-            Preconditions.checkNotNull(item, "iterable cannot contain null elements");
-        }
-
+        set.forEach(ConjureCollections::checkNotNullElement);
```

Comment:
Fwiw, LinkedHashSet does not seem to override Iterable's default forEach implementation, so we'll still create the iterator here afaict